### ISSUE TITLE
ProxyNCA Loss optimizer step location

### ISSUE THREAD

**Issue by alpoler** [KevinMusgrave/powerful-benchmarker#70]
Hi Kevin, I want to train proxy nca loss again but I don't figure out where I append loss optimizer to pay attention proxy weight during training.  Where should I append loss optimizer and its step to run training correctly ? @KevinMusgrave

**Comment by KevinMusgrave** [comment]
Add it to the [optimizers config](https://github.com/KevinMusgrave/powerful-benchmarker/blob/master/src/powerful_benchmarker/configs/config_optimizers/default.yaml) as ```metric_loss_optimizer```:
```
optimizers:
  metric_loss_optimizer:
    Adam:
      lr: 0.01
```

Or at the command line:
```
--optimizers {metric_loss_optimizer: {Adam: {lr: 0.01}}}
```

**Comment by alpoler** [comment]
Do you use the same lr and optimizer in your experiment mentioned above ? In addition, I also add weight regularizer to avoid trivial weight values (like 0 ). For this, I add this parameters to powerful benchmarker. 
`
 --loss_funcs~OVERRIDE~ {metric_loss: {ProxyNCALoss: {num_classes: 200, embedding_size: 128, weight_regularizer: {LpRegularizer : {}, weight_reg_weight=-1}}.`

Are they correct ?

**Comment by KevinMusgrave** [comment]
See this table for some hyperparameters that worked with RMSprop: https://kevinmusgrave.github.io/powerful-benchmarker/papers/mlrc/#optimal-hyperparameters. The momentum was 0.9 and weight decay was 0.0001.

There's a small mistake in your command line flag. Here's a corrected version:
```
--loss_funcs~OVERRIDE~ {metric_loss: {ProxyNCALoss: {num_classes: 200, embedding_size: 128, weight_regularizer: {LpRegularizer : {}}, weight_reg_weight: -1}}
```

Also by the way, you can leave out the ```num_classes``` argument and the benchmarker will automatically supply it with the number of training set classes.
